src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/MethodLengthCheck.java

```diff
@@ -273,24 +271,25 @@ private static int getLengthOfBlock(DetailAST openingBrace, DetailAST closingBra
     private static int countUsedLines(DetailAST ast) {
         final Deque<DetailAST> nodes = new ArrayDeque<>();
         nodes.add(ast);
-        final Set<Integer> usedLines = new HashSet<>();
+        final BitSet usedLines = new BitSet();
+        final int startLineNo = ast.getLineNo();
         while (!nodes.isEmpty()) {
             final DetailAST node = nodes.removeFirst();
-            final int lineNo = node.getLineNo();
+            final int lineNo = node.getLineNo() - startLineNo;
```

Comment:
Isn't this really number of lines not line number?

Reply:
This is a nano-optimization. If a method starts in line 10000 and ends on line 10010, we count them as 0 and 10, to avoid huge bit sets full or zeros. A `Bitset` is always zero-based, and here we make it zero-based.

Reply:
it can be named `lineIndex` in this case to avoid confusion

Reply:
done